File: server/php/unsubscribe-user-from-list.php
<?php
require_once __DIR__ . '/vendor/autoload.php';
use Courier\Client;
use Dotenv\Dotenv;

$dotenv = Dotenv::createImmutable(__DIR__ . '/..');
$dotenv->safeLoad();

$api_key = $_ENV['COURIER_API_KEY'] ?? '';
$list_id = $_ENV['COURIER_UNSUBSCRIBE_USER_FROM_LIST_LIST_ID'] ?? '';
$user_id = $_ENV['COURIER_UNSUBSCRIBE_USER_FROM_LIST_USER_ID'] ?? '';

$client = new Client(apiKey: $api_key);

$client->lists->subscriptions->unsubscribeUser($user_id, listID: $list_id);

File: server/kotlin/UnsubscribeUserFromList.kt
import com.courier.client.CourierClient
import com.courier.client.okhttp.CourierOkHttpClient
import com.courier.models.lists.subscriptions.SubscriptionUnsubscribeUserParams

fun main() {
    val apiKey = EnvLoader.getEnv("COURIER_API_KEY")
    val listId = EnvLoader.getEnv("COURIER_UNSUBSCRIBE_USER_FROM_LIST_LIST_ID")
    val userId = EnvLoader.getEnv("COURIER_UNSUBSCRIBE_USER_FROM_LIST_USER_ID")

    val client = CourierOkHttpClient.builder()
        .apiKey(apiKey)
        .build()

    val params = SubscriptionUnsubscribeUserParams.builder()
        .listId(listId)
        .userId(userId)
        .build()

    client.lists().subscriptions().unsubscribeUser(params)
}

